pkg/kubectl/resource_printer.go

```diff
@@ -599,6 +615,11 @@ func printPod(pod *api.Pod, w io.Writer, options PrintOptions) error {
 func printPodBase(pod *api.Pod, w io.Writer, options PrintOptions) error {
 	name := pod.Name
 	namespace := pod.Namespace
+	kind := options.KindName
```

Comment:
You can get the kind from the object. All top-level api objects have an implicit `unversioned.TypeMeta`, so you should be able to determine the kind or alias by writing a helper function like
```
func getKindFor(obj *unversioned.TypeMeta) string {
	kindName := obj.GroupVersionKind().Kind
	if alias, ok := ResourceShortFormFor(kindName); ok {
				kindName = alias
	} else if kindName == "" {
		kindName = "unknown"
	}
	return kindName
}
```

I suggest you do something like the above to determine the kind name and alias inside each `print` call. You do still have to pass a `WithKind` bool in options, but that's reasonable.